### ISSUE TITLE
No support for RC versions of node

### ISSUE THREAD

**Issue by MylesBorins** [yarnpkg/yarn#2355]
**Do you want to request a *feature* or report a *bug*?**
Bug

**What is the current behavior?**
yarn errors out saying that node is an incompatible version when using an rc version.

**If the current behavior is a bug, please provide the steps to reproduce.**
```
NVM_NODEJS_ORG_MIRROR=https://nodejs.org/download/rc  nvm install v6.9.3-rc.2
git clone https://github.com/facebook/jest
cd jest
yarn install
```

**What is the expected behavior?**
yarn should support engines that include an rc such as `v6.9.3-rc.2`

**Please mention your node.js, yarn and operating system version.**
node: v6.9.3-rc.2
yarn: 0.18.1
os: osx 10.10.5

**Comment by Daniel15** [comment]
Could you please provide the error message you're seeing?

**Comment by destroyerofbuilds** [comment]
```bash
$ node --version
v6.10.0-rc.1
```

```bash
$ cat package.json | grep "node"
    "node": "^6.9.0 || ^4.2.0 || ^0.12.0",
```

```bash
yarn --instal

...

error webpack@2.2.0: The engine "node" is incompatible with this module. Expected version ">=4.3.0 <5.0.0 || >=5.10".

error Found incompatible module
info Visit https://yarnpkg.com/en/docs/cli/install for documentation about this command.
```

Fails on `webpack`. I would expect their range to work with the 6.10 release candidate. This may not be a `yarn` issue, but wanted to post this here since I'm testing a Node release candidate.

**Comment by hbetts** [comment]
Cross linking this to #1102

**Comment by gsklee** [comment]
@bestander close

Can't do anything here because this is exactly how [`node-semver`](https://github.com/npm/node-semver) works; eg. if you require a `react` version `>=15`, you will not get version `16.0.0-alpha.13`. You should bring this issue to `node-semver` instead.

**Comment by bestander** [comment]
Actually Yarn decided to break semver here after multiple requests from the community.
This should be implemented in 0.25
